### PR TITLE
KEA resiliency

### PR DESCRIPTION
**Pull request by alvarez3-hpe**
## Summary and Scope

This makes resiliency changes to cray-dhcp-kea. This includes:
- Upgrading KEA to v2.4.0.
- Upgrading cray-service chart to 10.0.x.
- Scaling KEA pods up to 3 replicas.
- Changing KEA to use postgres backend.
- Split dhcp-helper loop off to a cron job.
- dhcp-helper updates the backup configmap to trigger kea config-reloads.
- Added init job to do initial database setup and KEA configuration.
- Changed KEA's IP address allocation algorithm to `random` to reduce instances of contention when handing out IP addresses.

## Issues and Related PRs

* Resolves [CASMHMS-6069](https://jira-pro.it.hpe.com:8443/browse/CASMHMS-6069)

### Tested on:

  * surtur
  * mug

### Test description:

- Was upgrade tested? Yes
- Was downgrade tested? Yes

Ran a few performance tests on mug. These performance tests were executed using a KEA provided tool, `perfdhcp`:

- From 1 compute node to 1 of 3 KEA pods with perfdchp acting like 100 clients sending 2000 requests per second. Shows that one pod can handle ~1000 DORA
- From 2 nodes to 2 of 3 KEA pods. Result showed that the per pod performance did not diminish when more than 1 pod is stressed. Shows that each cray-dhcp-kea pod can handle ~1000 DORA at the same time.

Needs further qualification on a large system.

## Risks and Mitigations

Without large scale testing, it is difficult to capture all edge cases. The old chart will be shipped alongside the new chart should it need to be reverted.

## Pull Request Checklist

- [x] Version number(s) incremented, if applicable
- [x] Copyrights updated
- [x] License file intact
- [x] Target branch correct
- [x] CHANGELOG.md updated
- [x] Testing is appropriate and complete, if applicable
- [x] [HPC Product Announcement](https://cray.slack.com/archives/C026TVCSXLH) prepared, if applicable

### CODE REVIEW

File: kubernetes/cray-dhcp-kea/files/startup-dhcp.sh [hunk 1]
```diff
@@ -7,6 +7,17 @@ BACKUP_CONFIG_PATH=/srv/kea/backup/
...
+#  check username and passward are empty
+#  this can happen if the pod restarts
+if [ -z "$DHCP_DBUSER" ]
+then
+      DHCP_DBUSER=$(cat /secrets/postgres/dhcpdsuser/username)
+fi
+if [ -z "$DHCP_DBPASS" ]
+then
+      DHCP_DBPASS=$(cat /secrets/postgres/dhcpdsuser/password)
+fi
```
Comment (rsjostrand-hpe) [inline kubernetes/cray-dhcp-kea/files/startup-dhcp.sh:19]: What are these env variables being used for? Looking at your changes to this file I don't see where they are used, unless I am missing something.

File: kubernetes/cray-dhcp-kea/files/startup-init.sh [hunk 2]
```diff
@@ -0,0 +1,80 @@
+#!/bin/bash
+
```
Comment (rsjostrand-hpe) [inline kubernetes/cray-dhcp-kea/files/startup-init.sh:2]: I think you need a MIT License header.

File: kubernetes/cray-dhcp-kea/files/startup-dhcp.sh (hunk 1 above)
Comment (alvarez3-hpe) [reply kubernetes/cray-dhcp-kea/files/startup-dhcp.sh:19]: They are used inside dhcp_helper

**Review by rsjostrand-hpe — DISMISSED** [review]

File: kubernetes/cray-dhcp-kea/values.yaml [hunk 3]
```diff
@@ -188,9 +242,26 @@ crayDhcpKeaCtrlAgentConfig:
         output_options:
           - output: stdout
         severity: ERROR
-global:
-  chart:
-    name: cray-dhcp-kea
-    version: 0.10.25
-  appVersion: 0.10.25
 
+cray-postgresql:
```
Comment (mjendrysik-hpe) [inline kubernetes/cray-dhcp-kea/values.yaml:246]: If you want the db backed up, you will need to add that here.

Something like this:
    enableLogicalBackup: true
    logicalBackupSchedule: "10 0 * * *"  # Once per day at 12:10AM

Comment (alvarez3-hpe) [reply kubernetes/cray-dhcp-kea/values.yaml:246]: We don't want to back it up

**Review by mjendrysik-hpe — APPROVED** [review]

**Review by rsjostrand-hpe — APPROVED** [review]

**Pull request merged**